posthog/temporal/ai/session_summary/summarize_session.py

```diff
@@ -384,6 +393,115 @@ async def run(self, inputs: SingleSessionSummaryInputs) -> None:
             )
 
 
+async def ensure_llm_single_session_summary(inputs: SingleSessionSummaryInputs):
+    if inputs.video_validation_enabled == "full":
+        # Full video-based summarization: analyze video directly with Gemini
+        retry_policy = RetryPolicy(maximum_attempts=3)
+        trace_id = temporalio.workflow.info().workflow_id
+
+        # Convert inputs to video workflow format
+        video_inputs = VideoSummarySingleSessionInputs(
+            session_id=inputs.session_id,
+            user_id=inputs.user_id,
+            user_distinct_id_to_log=inputs.user_distinct_id_to_log,
+            team_id=inputs.team_id,
+            redis_key_base=inputs.redis_key_base,
+            model_to_use="gemini-2.5-flash",  # Default model for video analysis
+            extra_summary_context=inputs.extra_summary_context,
+        )
+
+        # Activity 1: Export full session video
+        asset_id = await temporalio.workflow.execute_activity(
+            export_session_video_activity,
+            video_inputs,
+            start_to_close_timeout=timedelta(minutes=300),
+            retry_policy=retry_policy,
+        )
+
+        # Activity 2: Upload full video to Gemini (single upload)
+        uploaded_video = await temporalio.workflow.execute_activity(
+            upload_video_to_gemini_activity,
+            args=(video_inputs, asset_id),
+            start_to_close_timeout=timedelta(minutes=10),
+            retry_policy=retry_policy,
+        )
+
+        # Calculate segment specs based on video duration
+        # (We ignore the first couple of seconds, as they often include malformed frames)
+        analysis_duration = uploaded_video.duration - SESSION_VIDEO_RENDERING_DELAY
+        num_segments = int(analysis_duration / SESSION_VIDEO_CHUNK_DURATION_S)  # Number of segments is floored
+        segment_specs = [
+            VideoSegmentSpec(
+                segment_index=i,
+                start_time=SESSION_VIDEO_RENDERING_DELAY + i * SESSION_VIDEO_CHUNK_DURATION_S,
+                end_time=SESSION_VIDEO_RENDERING_DELAY
+                + (
+                    min((i + 1) * SESSION_VIDEO_CHUNK_DURATION_S, analysis_duration)
+                    if i < num_segments - 1
+                    else analysis_duration  # Final segment always reaches the end - a 28s chunk is preferable to a 2s one!
+                ),
+            )
+            for i in range(num_segments)
+        ]
+
+        # Activity 3: Analyze all segments in parallel
+        segment_tasks = [
```

Comment:
I would suggest adding `Semaphore` with limit of 10 or smth, as we can easily hit RPM/TPM.